python/ray/tune/tune.py

```diff
@@ -264,6 +265,18 @@ def run(run_or_experiment,
 
     for i, exp in enumerate(experiments):
         if not isinstance(exp, Experiment):
+            if callable(exp) and detect_checkpoint_function(exp):
+                if checkpoint_at_end:
+                    raise ValueError(
+                        "'checkpoint_at_end' cannot be used with a "
+                        "checkpointable function. You can specify "
+                        "and register checkpoints within "
+                        "your trainable function.")
+                if checkpoint_freq:
+                    raise ValueError(
+                        "'checkpoint_freq' cannot be used with a "
+                        "checkpointable function. You can specify checkpoints "
+                        "within your trainable function.")
```

Comment:
I think this belongs a couple lines ahead at 263

Reply:
run_or_experiment is only a list if it is a list of experiments.

Reply:
actually I think the solution is to revert back to the original code but to remove line 280 from tune.py. The experiment gets registered anyways in Experiment.__init__ so I don't see why we need to do it in tune.py. This way when we create the Experiment we pass in the function instead of a string so callable(run) returns true.